lib/generateClass.js

```diff
@@ -205,6 +213,65 @@ function writeFromJson(def, cw) {
   cw.ln('return inst;');
 }
 
+/**
+ * Generates a JSON serializer for the element
+ * @param {DataElement} def - The definition of the SHR element to generate a serializer for
+ * @param {CodeWriter} cw - The CodeWriter instance to use during generation
+ * @private
+ */
+function writeToJson(def, cw) {
+  // If the element is an Entry, put those fields on the JSON object first
+  if (def.isEntry) {
+    cw.ln(`const inst = this._entryInfo.toJSON();`);
+  } else if (def.identifier.name !== 'EntryType') {
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${def.identifier.name}`;
+    cw.ln(`const inst = { 'shr.base.EntryType': {'Value': '${url}'}};`);
+  } else {
+    cw.ln(`const inst = {};`);
+  }
+
+  if (def.value !== undefined) {
+    cw.bl('if (this.value !== undefined)', () => {
+      if (def.value instanceof ChoiceValue) {
+        // Choices get a special treatment
+        cw.ln(`inst['Value'] = typeof this.value.toJSON === 'function' ? this.value.toJSON() : this.value;`);
+        
+      } else if (def.value instanceof RefValue) {
+        // References get a special treatment too
+        cw.bl(`if (this.${toSymbol(def.value.identifier.name)} !== undefined)`, () => {
+          if (def.value.card.max > 1 || def.value.card.isMaxUnbounded) {
+            cw.ln(`inst['Value'] = this.value.map(f => f.toJSON());`);
+          } else {
+            cw.ln(`inst['Value'] = this.value.toJSON();`);
+          }
+        });
+        
+      } else if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
+        cw.ln(`inst['Value'] = this.value;`);
+      } else {
+        cw.ln(`inst['Value'] = this.value.toJSON();`);
+      }
+    });
+    
+  }
+
+  for (const field of def.fields) {
+    // Don't repeat properties that were purely inherited (without overriding).  Although overrides actually don't
+    // affect the definitions now, they may in the future, so we'll leave them in
+    if (field.inheritance !== INHERITED) {
+      cw.bl(`if (this.${toSymbol(field.identifier.name)} !== undefined)`, () => {
```

Comment:
This line actually crashed `shr-cli` when we run it against the SHR spec.  This is because `field` can also be an instance of `TBD` -- in which case it has no identifier.  You should probably just skip `TBD` fields.